File: app/src/main/java/com/themovielist/repository/favorite/MovieDAO.java
package com.themovielist.repository.favorite;


import android.database.Cursor;

import com.themovielist.model.MovieModel;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface MovieDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(MovieModel movieModel);

    @Query("SELECT * FROM " + MovieContract.MovieEntry.TABLE_NAME)
    Cursor selectAll();

    @Query("SELECT * FROM " + MovieContract.MovieEntry.TABLE_NAME + " WHERE " + MovieContract.MovieEntry._ID + " = :id")
    Cursor selectById(int id);

    @Query("DELETE FROM " + MovieContract.MovieEntry.TABLE_NAME + " WHERE " + MovieContract.MovieEntry._ID + " = :id")
    int deleteById(int id);
}

File: app/src/main/java/com/themovielist/repository/favorite/MovieDAO.kt
package com.themovielist.repository.favorite

import androidx.room.*
import com.themovielist.model.entity.MovieEntityModel
import com.themovielist.model.entity.MovieFavoriteWatchedEntityModel

@Dao
interface MovieDAO {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(movieEntityModel: MovieEntityModel)

    @Delete
    suspend fun delete(movieEntityModel: MovieEntityModel)

    @Query("SELECT id, is_favorite, is_watched from ${MovieEntityModel.TABLE_NAME}")
    suspend fun getFavoriteWatchedMovieList(): List<MovieFavoriteWatchedEntityModel>
}